Lib/asyncio/runners.py

```diff
@@ -0,0 +1,48 @@
+__all__ = 'run',
+
+from . import coroutines
+from . import events
+
+
+def run(main, *, debug=False):
+    """Run a coroutine.
+
+    This function runs the passed coroutine, taking care of
+    managing the asyncio event loop and finalizing asynchronous
+    generators.
+
+    This function cannot be called when another asyncio event loop is
+    running in the same thread.
+
+    If debug is True, the event loop will be run in debug mode.
+
+    This function always creates a new event loop and closes it at the end.
+    It should be used as a main entry point for asyncio programs, and should
+    ideally only be called once.
+
+    Example:
+
+        async def main():
+            await asyncio.sleep(1)
+            print('hello')
+
+        asyncio.run(main())
+    """
+    if events._get_running_loop() is not None:
+        raise RuntimeError(
+            "asyncio.run() cannot be called from a running event loop")
+
+    if not coroutines.iscoroutine(main):
+        raise ValueError("a coroutine was expected, got {!r}".format(main))
+
+    loop = events.new_event_loop()
+    try:
+        events.set_event_loop(loop)
```

Comment:
`set_event_loop()` cannot fail (it has a couple asserts but we can neglect this fact).

Reply:
Sure it can. It calls `policy.set_event_loop()`, and given the fact that users can provide their own buggy policies, we can't really say that `set_event_loop` is 100% safe.

Reply:
It's a flag of broken third party loop implementation, not user code problem.
Buggy loop can do weird things, I expect asyncio will be broken somehow anyway.

Reply:
What's your point?

The point of this code is to always close the `loop`, no matter what.  What benefit is there in moving `set_event_loop()` one line up?

Reply:
Anyways, if you think that the code will read better if `set_event_loop` is outside of the `try` block I can move it. We call `set_event_loop` in the `finally` block along with the `loop.close()`, so as you say: if these things are broken then nothing will actually work anyways.

Reply:
Well, it is very minor thing.
From my perspective wrapping `loop.run_until_complete(main)` only increases code readability a little but I can live with current implementation too.

Reply:
I'll merge as-is then.